multipaz/src/androidMain/kotlin/org/multipaz/mdoc/transport/BlePeripheralManagerAndroid.kt

```diff
@@ -539,20 +539,26 @@ internal class BlePeripheralManagerAndroid: BlePeripheralManager {
         device = null
         advertiser?.stopAdvertising(advertiseCallback)
         advertiser = null
-        gattServer?.removeService(service)
-        gattServer?.close()
-        gattServer = null
-        service = null
-        l2capServerSocket?.close()
-        l2capServerSocket = null
-        incomingMessages.close()
         l2capSocket?.let {
-            CoroutineScope(Dispatchers.IO).launch() {
-                delay(5000)
+            CoroutineScope(Dispatchers.IO).launch()  {
+                try {
+                    it.outputStream?.flush()
+                    delay(5000)
+                } catch (e: Exception) {
+                    Logger.w("LongfellowZkSystem", "Error flushing", e)
+                }
+
+                gattServer?.removeService(service)
+                gattServer?.close()
+                gattServer = null
+                service = null
+                l2capServerSocket?.close()
+                l2capServerSocket = null
+                incomingMessages.close()
                 it.close()
+                l2capSocket = null
```

Comment:
This here is definitely not right, it won't call e.g. gattServer.removeService() if we're not using l2cap sockets.